### ISSUE TITLE
Deprecated warnings in ansible 2.8 with v1.5

### ISSUE THREAD

**Issue by trickert76** [lae/ansible-role-proxmox#57]
Hi - I love this project - it makes my life much more easier. But there are at least some very minor but later more major warnings. With Ansible 2.8 some variable managing changes - so this should be easy be updated to avoid later problems with newer Ansible releases

TASK [lae.proxmox : Ensure this host is in the group specified] **********************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

...

TASK [lae.proxmox : Perform system upgrades] *****************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_run_system_upgrades as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the 
future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

...

TASK [lae.proxmox : Stage ZFS packages if ZFS is enabled] ****************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_zfs_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. Also 
see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False
 in ansible.cfg.
skipping: [ac-sh002]

...

TASK [lae.proxmox : Remove subscription check wrapper function in web UI] ************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_remove_subscription_warning as a bare variable, this behaviour will go away and you might need to add |bool to the expression in 
the future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
ok: [ac-sh002]

TASK [lae.proxmox : Check for kernel update] *****************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_reboot_on_kernel_update as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the 
future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
ok: [ac-sh002]

TASK [lae.proxmox : Reboot for kernel update] ****************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_reboot_on_kernel_update as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the 
future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Wait for server to come back online] *****************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_reboot_on_kernel_update as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the 
future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Remove old Debian/PVE kernels] ***********************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_remove_old_kernels as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future.
 Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
ok: [ac-sh002]

TASK [lae.proxmox : Load ZFS module live] ********************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_zfs_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. Also 
see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False
 in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Load ZFS module on init] *****************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_zfs_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. Also 
see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False
 in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Copy ZFS modprobe configuration] *********************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_zfs_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. Also 
see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False
 in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Configure email address for ZFS event daemon notifications] ******************************************************************************************
[DEPRECATION WARNING]: evaluating pve_zfs_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. Also 
see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False
 in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Lookup cluster information] **************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Identify if the host is already part of a cluster] ***************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Identify all clusters that the hosts in the specified group may be in] *******************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002] => (item=ac-sh002) 

TASK [lae.proxmox : Ensure that hosts found are not in multiple existing clusters] ***************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Ensure that, if we find an existing cluster, that it matches the specified cluster name] *************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Initialize a Proxmox cluster] ************************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Wait for quorum on initialization node] **************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_cluster_enabled as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [lae.proxmox : Install Proxmox Let's Encrypt post-hook script] ******************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_ssl_letsencrypt as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

TASK [Request Let's Encrypt certificate for ...] ************************************************************************************************************
[DEPRECATION WARNING]: evaluating pve_ssl_letsencrypt as a bare variable, this behaviour will go away and you might need to add |bool to the expression in the future. 
Also see CONDITIONAL_BARE_VARS configuration toggle.. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [ac-sh002]

Sorry - but I didn't check the dev branch for it

**Comment by trickert76** [comment]
If I would know (I need to learn how to use ansible-galaxy) I would think, that added the |bool would be enough to get rid of the warnings. But I'm not sure if this works with Ansible 2.4 or less (I'm new to Ansible).

Example:

```
- name: Check for kernel update
  collect_kernel_info:
    lookup_packages: false
  register: _pve_kernel_update
  when: pve_reboot_on_kernel_update

```
shoud be changed to:
```
- name: Check for kernel update
  collect_kernel_info:
    lookup_packages: false
  register: _pve_kernel_update
  when: pve_reboot_on_kernel_update|bool
```

**Comment by lae** [comment]
Versions older than Ansible 2.4 are EOL so I'm not going to worry about supporting those versions. Also, master is currently in sync with develop, so you're probably already using the most current version (1.5.0) I'd assume.

Does appending the `when` conditions with a ` | bool` suffice (whitespace added here for consistency with existing codebase)? It sounds like it should. Do you think you could submit a pull request?

**Comment by trickert76** [comment]
I'm working on it.

**Comment by trickert76** [comment]
ok, I tested it now with my patch and I don't get any warnings.

**Comment by lae** [comment]
Resolved in #58
